jena-kafka-connector/src/test/java/org/apache/jena/kafka/utils/TestEnvVariables.java

```diff
@@ -0,0 +1,141 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockitoExtension.class)
+public class TestEnvVariables {
+
+    @Test
+    public void testPlainValue() {
+        // given, when, then
+        String result = EnvVariables.checkForEnvironmentVariableValue("testConfig", "plainValue");
+        assertEquals("plainValue", result);
+    }
+
+    @Test
+    public void testNullValue() {
+        // given, when, then
+        String result = EnvVariables.checkForEnvironmentVariableValue("testConfig", null);
+        assertNull(result);
+    }
+
+    @Test
+    public void testEnvValue() {
+        // given
+        String envVarName = "TEST_ENV_VAR";
+        String envVarValue = "envValue";
+        Map<String, String> env = new HashMap<>();
+        env.put(envVarName, envVarValue);
+        setEnvironmentVariables(env);
+        // when
+        String result = EnvVariables.checkForEnvironmentVariableValue("testConfig", "env:" + envVarName);
+        // then
+        assertEquals(envVarValue, result);
+    }
+
+    @Test
+    public void testEnvValueWithBraces() {
+        String envVarName = "TEST_ENV_VAR";
+        String envVarValue = "envValue";
+        Map<String, String> env = new HashMap<>();
+        env.put(envVarName, envVarValue);
+        setEnvironmentVariables(env);
+
+        String result = EnvVariables.checkForEnvironmentVariableValue("testConfig", "env:{" + envVarName + "}");
+        assertEquals(envVarValue, result);
+    }
+
+    @Test
+    public void testEnvValueWithDefault() {
+        String envVarName = "TEST_ENV_VAR_DEFAULT";
+        String defaultValue = "defaultValue";
+
+        String result = EnvVariables.checkForEnvironmentVariableValue("testConfig", "env:{" + envVarName + ":" + defaultValue + "}");
+        assertEquals(defaultValue, result);
+    }
+
+    @Test
+    public void testEnvValueWithDefaultAndEnvSet() {
+        String envVarName = "TEST_ENV_VAR";
+        String envVarValue = "envValue";
+        String defaultValue = "defaultValue";
+        Map<String, String> env = new HashMap<>();
+        env.put(envVarName, envVarValue);
+        setEnvironmentVariables(env);
+
+        String result = EnvVariables.checkForEnvironmentVariableValue("testConfig", "env:{" + envVarName + ":" + defaultValue + "}");
+        assertEquals(envVarValue, result);
+    }
+
+    @Test
+    public void testEnvValueNotSet() {
+        String envVarName = "MISSING_TEST_ENV_VAR";
+        assertThrows(JenaException.class, () -> {
+            EnvVariables.checkForEnvironmentVariableValue("testConfig", "env:{" + envVarName + "}");
+        });
+    }
+
+    @Test
+    public void testInvalidEnvExpression() {
+        assertThrows(JenaException.class, () -> {
+            EnvVariables.checkForEnvironmentVariableValue("testConfig", "env:{INVALID_EXPRESSION");
+        });
+    }
+
+    private void setEnvironmentVariables(Map<String, String> newenv) {
```

Comment:
Elsewhere on one of Graham's PRs there was a somewhat more elegant approach to this using the `SystemStubsExtension` - https://github.com/Telicent-io/edh-pbac-federation-filter/pull/18 - that might be worth reusing?

Reply:
`Lib.getenv` abstracts accessing environment variables - it also looks in system properties. 
System properties override environment variables so embedded use is more flexible (including tests).

Reply:
@TelicentPaul If there's a preexisting abstraction as @afs points then might be better to just reuse that are fake the configuration by setting and unsetting System Properties if that isn't too much refactoring?

Reply:
True - it's for a slightly different use case. Allowing the program to "override" environment variables either temporarily (e.g. in a test) or because the environment variable is the default and program doesn't want it.

Here either works.